File: day_i/part1/part1.c
#if 0
gcc -o $0.out $0 -lm && $0.out $@
rm $0.out &>/dev/null
exit
#endif
#include <stdio.h>
#include <stdlib.h>

#define println(x, ...) printf(x "\n", ##__VA_ARGS__)

int main(int argc, char **argv) {
  if (argc < 2) {
    println("Please provide input File");
    return -1;
  }
  char *filename = (++argv)[0];
  FILE *fp = fopen(filename, "r");
  int answer = 0, temp[2] = {0}, f = -1;
  char c;
  while (c != EOF) {
    c = fgetc(fp);
    if (c >= '0' && c <= '9') {
      if (f < 1) {
        f++;
      }
      temp[f] = c - '0';
    } else if (c == 10 || c == EOF) {
      answer += temp[0] * 10 + temp[f];
      f = -1;
    }
  }
  println("The answer is [%d]", answer);
  return 0;
}

File: day_i/part1/part1.go
package main

import (
	"fmt"
	"os"
)

func main() {
	if len(os.Args) < 2 {
		fmt.Print("Please Provide the File to read")
	}
	_fil, _err := os.Open(os.Args[1])
	if _err != nil {
		return
	}
	var b []byte = []byte{' '}
	var answer int
	f := -1
	var temp [2]byte
	var count int = 1
	for count > 0 {
		_, err := _fil.Read(b)
		if b[0] >= '0' && b[0] <= '9' {
			if f < 1 {
				f++
			}
			temp[f] = b[0] - '0'
		}
		if b[0] == '\n' {
			answer += int(temp[0]*10 + temp[f])
			f = -1
		}
		if err != nil{
			answer += int(temp[0]*10 + temp[f])
			break
		}
	}
	print("The Answer is ",answer,"\n")
}
